### ISSUE TITLE
Description nullable on old installations but not nullable on fresh ones

### ISSUE THREAD

**Issue by mikaelmello** [duplicati/duplicati#3428]
## Description

Since the recent addition of a description field for backups, if you upgrade from an older version, by migrating the `Backup` schema is altered and the `Description` column is added with `NULL` as the default value.

However, the schema itself (used on new installations) declares `Description` as `NOT NULL`, hence two problems arrive:

1. There is an inconsistency that can only be fixed by resetting the database. Should we make the `Column` column nullable? If not, the migration should probably add the new column as nullable, update all entries to an empty string and make the column not nullable. Is that a good approach?

2. There is nothing handling a missing description error. An error while validating the insertion into the db is propagated all the way to the UI, displaying this:

![image](https://user-images.githubusercontent.com/17488630/46957979-2ece0100-d06f-11e8-8375-08808d365f16.png)

I'm not sure how you guys would want it fixed, so I don't have a PR ready.

**Comment by Pectojin** [comment]
Good catch! I'm thinking the best `NOT NULL` value would be an empty string when talking about descriptions. That way we don't need to deal with null exceptions other places in the application since it should always be a string.

Your described approach of upgrading in two steps seems like the best approach.
